client/src/main/java/io/pravega/client/stream/impl/EventPointerInternal.java

```diff
@@ -34,21 +34,21 @@
      *
      * @return a segment object
      */
-    abstract Segment getSegment();
+    public abstract Segment getSegment();
 
     /**
      * Get the start offset of the event.
      *
      * @return the start offset for this event
      */
-    abstract long getEventStartOffset();
+    public abstract long getEventStartOffset();
 
     /**
      * Get the length of the event.
      *
      * @return the event length.
      */
-    abstract int getEventLength();
+    public abstract int getEventLength();
```

Comment:
These should not be public. It exposes these to the user. If this is needed in another class you can do it via indirection by having an implementation of the method being invoked located in the the segment.impl package.